### DISCUSSION TITLE
Auth guard "user_id"

### DISCUSSION THREAD

**Discussion by backspacerhino** [adonisjs/core#1285] [Help] [answered]
Hi,

When I use 

```js
const token = await auth.use('someguard').generate(account, {
  expiresIn: '10 days',
})
```

I get error for my `account_api_tokens` that has "account_id" but generate() is trying to insert "user_id"

Is there a way for me to specify the foreign key name instead of default "user_id"

**Comment by thetutlage** [comment, marked as answer]
Any reasons for a using a different foreign key name? The `user_id` is required to know for which user the token is generated

**Reply by backspacerhino** [reply]
There are two different entities in my application, users and accounts. There are also a specific hardware products that can authenticate to server. It would be misleading to call accounts or hardware products users when there is also an entity "User".

**Reply by thetutlage** [reply]
That makes perfect sense. Lemme add it to the config

**Reply by backspacerhino** [reply]
Thank you very much

**Comment by fnoquiq** [comment]
In adonis v5, just change the settings in 'config/auth.ts'

```ts

import Account from 'App/Models/Account'
import { AuthConfig } from '@ioc:Adonis/Addons/Auth'

const authConfig: AuthConfig = {
  guard: 'api',
  list: {
    api: {
      driver: 'oat',
      tokenProvider: {
        driver: 'database',
        table: 'api_tokens',
      },

      provider: {
        driver: 'lucid',

        identifierKey: 'id',

        uids: ['id'],

        model: Account,
      },
    },
  },
}
```
